src/server/typingsInstaller/typingsInstaller.ts

```diff
@@ -332,6 +338,9 @@ namespace ts.server.typingsInstaller {
                     if (this.log.isEnabled()) {
                         this.log.writeLine(`Installed typings ${JSON.stringify(scopedTypings)}`);
                     }
+
+                    this.writePackageVersion(cachePath);

```

Comment:
Hmm, yes I see how this could fail. The packages would only be refreshed if the project requests them, so I could be writing out the TS version before everything's been refreshed.

So the main scenario is: User uninstalls the latest version of TypeScript. Now they're left with .d.ts files that don't compile with the current version. There are other scenarios that could be solved but this is the main one.

The most straightforward solution is to blow away the cache (or devdependencies) entirely. But there are lots of scenarios where user can be switching TS versions frequently (opening different projects with different TS versions) and we don't want to be rendering the cache unusable in offline scenarios.

I am trying to strike a balance between
- Making sure user doesn't get stuck with broken .d.ts files (i.e. files that target a higher version of TS than currently running)
- Not blowing away the cache excessively
- Not making excessive npm calls

I'm beginning to think that having a separate cache subfolder per-version might be a way to go.